### PR TITLE
Fetch log details immediately upon receiving logs

### PR DESCRIPTION
**Pull request by kenzieschmoll**
This PR:

Fetches log details immediately upon creation. This eliminates the issue where we seemingly lose data because we are lazy loading log details, and by the time we load the details, the object we are trying to fetch may have already been garbage collected. Fixes https://github.com/flutter/devtools/issues/7575

This PR pulls over some of the changes from the logging v2 work that is behind a flag.

### CODE REVIEW

File: packages/devtools_app/lib/src/screens/logging/logging_controller.dart [hunk 1]
```diff
@@ -740,7 +741,10 @@ class LogData with SearchableDataMixin {
     this.isError = false,
     this.detailsComputer,
     this.node,
-  });
+  }) {
+    // Fetch details immediately on creation.
+    unawaited(compute());
```
Comment (elliette) [inline packages/devtools_app/lib/src/screens/logging/logging_controller.dart:746]: Is it possible `compute()` might throw? If so, maybe we should call `catchError`

**Review by elliette — APPROVED** [review]
LGTM with suggesion!

File: packages/devtools_app/lib/src/screens/logging/logging_controller.dart (hunk 1 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/screens/logging/logging_controller.dart]: Good idea! done.

File: packages/devtools_app/lib/src/screens/logging/_message_column.dart [hunk 2]
```diff
@@ -81,17 +85,38 @@ class MessageColumn extends ColumnData<LogData>
...
-            //  The current summary is generally the first 200 chars of details.
-            getDisplayValue(data),
-            Theme.of(context).regularTextStyle,
-          ),
-        ),
-        overflow: TextOverflow.ellipsis,
-        maxLines: 1,
+      return ValueListenableBuilder<bool>(
+        valueListenable: data.detailsComputed,
+        builder: (context, detailsComputed, __) {
```
Comment (bkonyi) [inline packages/devtools_app/lib/src/screens/logging/_message_column.dart:90]: Nit: s/__/_/

**Review by bkonyi — APPROVED** [review]

**Pull request merged**